ReactAndroid/src/main/java/com/facebook/react/uimanager/TouchTargetHelper.java

```diff
@@ -144,8 +148,19 @@ private static boolean isTransformedTouchPointInView(
       localX = localXY[0];
       localY = localXY[1];
     }
-    if ((localX >= 0 && localX < (child.getRight() - child.getLeft()))
-        && (localY >= 0 && localY < (child.getBottom() - child.getTop()))) {
+    float hitSlopTop = 0;
```

Comment:
We don't want to prioritize views with hitSlop. We just want to treat the views as bigger than they actually are so that you don't have to make your styles more complicated or wrap buttons with an extra View.

In [1] we would respect the ordering of the views - if Child1 is above Child2 it'd get the touch and vice versa.

For [2] this is the same behavior as when you extend Android touch areas with http://developer.android.com/reference/android/view/TouchDelegate.html. It is also consistent with the iOS portion of this PR.